frontend/src/components/ui/Select/Select.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useState, useRef, useEffect, forwardRef } from "react";
+import { cn } from "../../../lib/utils";
+import styles from "./Select.module.scss";
+
+export interface SelectOption {
+  value: string;
+  label: string;
+  disabled?: boolean;
+}
+
+export interface SelectProps {
+  options: SelectOption[];
+  value?: string;
+  onChange?: (value: string) => void;
+  placeholder?: string;
+  disabled?: boolean;
+  className?: string;
+  error?: boolean;
+  helperText?: string;
+
+/** Custom color overrides */
+  background?: string;      // e.g. "var(--clr-background)" or "#fff"
+  textColor?: string;       // e.g. "var(--clr-text)"
+  borderColor?: string;
+  accentColor?: string;
+  hoverColor?: string;
+}
+
+const Select = forwardRef<HTMLDivElement, SelectProps>(
```

Comment:
I wonder why we need a forwardRef here? What does it want to achieve?

I notice that the ref is currently exposing the outer container div, while there is another wrapper ref which I'm not sure what is it for. Can you please describe your decisions with more details?